openeo/rest/imagecollectionclient.py

```diff
@@ -970,8 +970,29 @@ def save_result(self, format: str = "GTIFF", options: dict = None):
             }
         )
 
-    def download(self, outputfile: str = None, format: str = "GTIFF", options: dict = None):
+    def download(self, outputfile: str = None, format: str = None, options: dict = None):
         """Download image collection, e.g. as GeoTIFF."""
+        if format == None:
+            dots = [i for i, ltr in enumerate(outputfile) if ltr == "."]
+            format_string = outputfile[dots[len(dots)-1]+1:len(outputfile)].upper()
+
+            if format_string == "TIFF":
```

Comment:
code tip: instead of this large if-elif-construct you can also define a dictionary mapping file extensions to formats, e.g.
```python
format_map = {
    "GTIFF": "GTIFF", "GEOTIFF": "GTIFF", "TIFF": "GTIFF", "TIF": "GTIFF",
    "JSON": "JSON",
    "NC": "NetCDF", "NETCDF": "NetCDF",
...
format = format_map[extension]
```

that's more compact and easier to maintain